docs/src/main/tut/cookbook.md

```diff
@@ -0,0 +1,650 @@
+    }
+
+    p
+  }
+}
+```
+
+### Server Sent Events
+
+Finch offers support for [Server Sent Events](server-sent-events) through the `finch-sse` sub-project.
+Server Sent Events are represented as `AsyncStream`s and streamed over the chunked HTTP transport.
+
+The `ServerSentEvent` case class caries an arbitrary `data` field and it's possible to encode any
+`ServerSentEvent[A]` for which `cats.Show[A]` is defined.
+
+In this example, every next second we stream instances of `java.util.Date` as server sent events on
+the `time` endpoint.
+
+NOTE: SSE requires `Cache-Control` to be disabled.
+
+```tut:silent
+import cats.Show
+import com.twitter.concurrent.AsyncStream
+import com.twitter.conversions.time._
+import com.twitter.finagle.Http
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util.{ Await, Future, Timer}
+import io.finch._
+import io.finch.sse._
+import java.util.Date
+
+implicit val showDate: Show[Date] = Show.fromToString[Date]
+
+implicit val timest: Timer = DefaultTimer.twitter
+
+def streamTime(): AsyncStream[ServerSentEvent[Date]] =
+  AsyncStream.fromFuture(
+    Future.sleep(1.seconds)
+          .map(_ => new Date())
+          .map(ServerSentEvent(_))
+  ) ++ streamTime()
+
+val time: Endpoint[AsyncStream[ServerSentEvent[Date]]] = get("time") {
+  Ok(streamTime())
+    .withHeader("Cache-Control" -> "no-cache")
+}
+
+//Await.ready(Http.server.serve(":8081", time.toServiceAs[Text.EventStream]))
+```
+
+### JSONP
+
+Not going into the details on why [JSONP considered insecure][insecure-jsonp], there is a Finagle
+filter `JsonpFilter` that can be applied to an HTTP service returning JSON to "upgrade" it to JSONP.
+
+Here is a small example on how to wire this filter with Finch's endpoint.
+
+```tut:silent
+import com.twitter.finagle.Http
+import com.twitter.finagle.http.filter.JsonpFilter
+import io.finch._
+import io.finch.circe._
+import io.circe.generic.auto._
+
+val endpoint: Endpoint[Map[String, String]] = get("jsonp") {
+  Ok(Map("foo" -> "bar"))
+}
+
+val service = endpoint.toServiceAs[Application.Json]
+
+// Http.server.serve(":8080", JsonpFilter.andThen(service))
+```
+
+`JsonpFilter` is dead simple. It checks the returned HTTP payload and if it's a JSON string, wraps
+it with a call to a function whose name is passed in the `callback` query-string param (and changes
+the content-type to `application/javascript` correspondingly). Using [HTTPie][httpie], this would look as:
+
+```
+$ http :8081/jsonp
+HTTP/1.1 200 OK
+Content-Encoding: gzip
+Content-Length: 39
+Content-Type: application/json
+
+{
+    "foo": "bar"
+}
+
+$ http :8080/jsonp?callback=myfunction
+HTTP/1.1 200 OK
+Content-Encoding: gzip
+Content-Length: 56
+Content-Type: application/javascript
+
+/**/myfunction({"foo":"bar"});
+```
+
+### OAuth2
+
+There is [finagle-oauth2](https://github.com/finagle/finagle-oauth2) server-side provider that is
+supported in Finch via the `finch-oauth2` package:
+
+*Authorize*
+```tut:silent
```

Comment:
By some reason, formatting is sort of weird here.  I played with it and figured that if we add a spacing after `Authorize` it works just great. I'm happy yo do that later in the upcoming commits, but you if want to fix this as well in your next diff that would awesome (just let me know).